pkg/logs/sender/sender_test.go

```diff
@@ -17,6 +19,56 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/logs/message"
 )
 
+type mockDestination struct {
+	errChan  chan bool
+	hasError bool
+}
+
+func newMockDestination() *mockDestination {
+	return &mockDestination{
+		errChan:  make(chan bool, 1),
+		hasError: false,
+	}
+}
+
+func (m *mockDestination) Send(payload []byte) error {
+	select {
+	case m.hasError = <-m.errChan:
+	default:
+	}
+
+	if m.hasError {
+		return errors.New("Test error")
+	}
+	return nil
+
+}
+func (m *mockDestination) SendAsync(payload []byte) {
+}
+
+type mockStrategy struct {
+	sendFailed chan bool
+}
+
+func newMockStrategy() *mockStrategy {
+	return &mockStrategy{
+		sendFailed: make(chan bool),
+	}
+
+}
+
+func (m *mockStrategy) Send(inputChan chan *message.Message, outputChan chan *message.Message, send func([]byte) error) {
+	for msg := range inputChan {
+		if send(msg.Content) == nil {
+			outputChan <- msg
```

Comment:
nit: To not indent twice, you can do a continue right after this line instead I think